testdata/generate_test_data/test_circles_onto.json

```diff
@@ -0,0 +1,86 @@
+    "project": {
+        "shortcode": "0700",
+        "shortname": "simcir",
+        "longname": "Very small ontology.",
+        "descriptions": {
+            "en": "Very small ontology."
+        },
+        "keywords": [
+            "test",
+            "testing"
+        ],
+        "ontologies": [
+            {
+                "name": "simcir",
+                "label": "Test ontology",
+                "comment": "To test the circular references.",
+                "properties": [
+                    {Å
+                        "name": "hasRichtext",
+                        "super": [
+                            "hasValue"
+                        ],
+                        "object": "TextValue",
+                        "labels": {
+                            "en": "hasRichtext"
+                        },
+                        "gui_element": "Richtext"
+                    },
+                    {
+                        "name": "hasResource",
+                        "super": [
+                            "hasLinkTo"
+                        ],
+                        "object": "Resource",
+                        "labels": {
+                            "en": "hasResource"
+                        },
+                        "gui_element": "Searchbox"
+                    },
+                    {
+                        "name": "hasSimpleText",
+                        "super": [
+                            "hasValue"
+                        ],
+                        "object": "TextValue",
+                        "labels": {
+                            "en": "hasSimpleText"
+                        },
+                        "gui_element": "SimpleText"
+                    }
+                ],
+                "resources": [
+                    {
+                        "name": "TestThing",
+                        "super": [
+                            "Resource"
+                        ],
+                        "labels": {
+                            "en": "TestThing",
+                            "rm": "TestThing in Rumantsch"
+                        },
+                        "cardinalities": [
+                            {
+                                "propname": ":hasResource",
+                                "cardinality": "0-n",
+                                "gui_order": 0
+                            },
+                            {
+                                "propname": ":hasRichtext",
+                                "cardinality": "0-n",
+                                "gui_order": 1
+                            },
+                            {
+                                "propname": ":hasSimpleText",
+                                "cardinality": "0-n",
+                                "gui_order": 2
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    }
+}
```

Comment:
missing final newline (which is considered good practice)

I have my IDE configured so that it adds that automatically, which I would recommend you do too